src/component/Sticky/Sticky.tsx

```diff
@@ -0,0 +1,250 @@
+import { onResetDragSelection, OnResetDragSelection } from '../../store/thunks';
+import * as styles from './Sticky.scss';
+
+type DragFunction = (event: DragEvent) => void;
+export const STICKY_SPEC_ID: string = 'sticky-container';
+
+export interface StickyPassedProps {
+    uuid: string;
+    sticky: StickyNote;
+    plumberClearDragSelection: () => void;
+    plumberRemove: (uuid: string) => void;
+    plumberDraggable: (
+        uuid: string,
+        start: DragFunction,
+        drag: DragFunction,
+        stop: DragFunction,
+        beforeDrag?: () => void
+    ) => void;
+}
+
+export interface StickyStoreProps {
+    definition: FlowDefinition;
+    updateSticky: UpdateSticky;
+    onResetDragSelection: OnResetDragSelection;
+}
+
+export type StickyProps = StickyPassedProps & StickyStoreProps;
+
+/**
+ * We have internal state to track as the user types so
+ * we can debounce updates to the store.
+ */
+interface StickyState {
+    title: string;
+    body: string;
+    color: string;
+    showConfirmation: boolean;
+}
+
+const COLOR_OPTIONS = {
+    yellow: styles.yellow,
+    blue: styles.blue,
+    green: styles.green,
+    purple: styles.purple,
+    gray: styles.gray
+};
+
+export class Sticky extends React.Component<StickyProps, StickyState> {
+    private dragging = false;
+    private ele: HTMLDivElement;
+    private debounceTextChanges: number;
+    private showConfirmation: number;
+
+    constructor(props: StickyProps & StickyStoreProps) {
+        super(props);
+        bindCallbacks(this, {
+            include: [/^on/, /^get/]
+        });
+
+        this.state = {
+            title: this.props.sticky.title,
+            body: this.props.sticky.body,
+            color: this.props.sticky.color,
+            showConfirmation: false
+        };
+    }
+
+    private onRef(ref: HTMLDivElement): HTMLDivElement {
+        return (this.ele = ref);
+    }
+
+    public componentDidMount(): void {
+        this.props.plumberDraggable(
+            this.props.uuid,
+            (event: DragEvent) => this.onDragStart(event),
+            (event: DragEvent) => this.onDrag(event),
+            (event: DragEvent) => this.onDragStop(event),
+            () => {
+                return true;
+            }
+        );
+    }
+
+    public componentWillUnmount(): void {
+        this.props.plumberRemove(this.props.uuid);
+
+        if (this.showConfirmation) {
+            window.clearTimeout(this.showConfirmation);
+        }
+
+        if (this.debounceTextChanges) {
+            window.clearTimeout(this.showConfirmation);
+        }
+    }
+
+    private onDragStart(event: DragEvent): void {
+        this.props.plumberClearDragSelection();
+        this.props.onResetDragSelection();
+    }
+
+    private onDrag(event: DragEvent): void {
+        // noop
+    }
+
+    private onDragStop(event: DragEvent): void {
+        // snap us to the same grid
+        const { left, top } = snapToGrid(event.finalPos[0], event.finalPos[1]);
```

Comment:
JW, since stickies can be placed anywhere on the canvas...aren't really restricted in any way...should we be snapping them to the grid? is that something you had considered? OCD ALERT: i played with dragging a sticky over a node (something i'd probably do if i were on a team sharing an account), and was disappointed that it was snapping to the canvas.

Reply:
If you check the todos, future plans here are to have notes play in the same space as nodes. ie, you can insert them in between nodes and they will reflow, can drag select and move together with nodes, etc

Reply:
But does that require them to snap? If so, why?

Reply:
Not sure why they would play differently than nodes in that case. Especially if you are inserting them in between nodes and reflowing.

Reply:
Yeah I guess that would be a req. How committed are we to putting those on the same plane as nodes? Would that be resolving an issue peeps are running into in the v1 editor?

Reply:
Pretty committed, just feels more natural being able to insert notes and the flow making room as necessary just as it does with the rest of the flow. That said, I don't see it as an essential thing out the gate.

Reply:
Yeah, fair. Guess it does make sense to really lean into the reflow paradigm here.